xds/src/main/java/io/grpc/xds/LocalityStore.java

```diff
@@ -249,30 +259,76 @@ public void updateLocalityStore(Map<XdsLocality, LocalityInfo> localityInfoMap)
         }
         newState = aggregateState(newState, childHelper.currentChildState);
       }
-      localityMap = Collections.unmodifiableMap(updatedLocalityMap);
 
-      updatePicker(newState, childPickers);
+      // append deactivated localities to the end of localityMap
```

Comment:
The ordering in the map doesn't matter. I would just say "Add deactivated localities to localityMap to keep track of them", so that the audience wouldn't wonder if the order is important.

Reply:
Done.